actions/task.go

```diff
@@ -0,0 +1,280 @@
+	}
+
+	tasks := models.Tasks{}
+	search := &models.Search{}
+
+	if err := c.Bind(search); err != nil {
+		return err
+	}
+
+	q := tx.Where("finished_at IS not null")
+
+	if search.Name != "" {
+		s := fmt.Sprintf("%%%v%%", search.Name)
+		q.Where("name like  ?", s)
+
+	} else if !search.Date.IsZero() {
+
+		q.Where("created_at::date = ?::date", search.Date)
+
+	}
+	if err := q.All(&tasks); err != nil {
+		return err
+	}
+
+	countInfoComplete := fmt.Sprintf("%v Complete Tasks", len(tasks))
+	taskInfo := tasks.InfoTask(uuid.FromStringOrNil(c.Param("task_id")))
+
+	c.Set("taskInfo", taskInfo)
+	c.Set("count", countInfoComplete)
+	c.Set("underline", tasks)
+	c.Set("tasks", tasks)
+
+	return c.Render(http.StatusOK, r.HTML("complete_table_tasks.plush.html"))
+}
+
+func Create(c buffalo.Context) error {
+
+	tx, ok := c.Value("tx").(*pop.Connection)
+	if !ok {
+		return errors.New("no transaction found")
+	}
+
+	task := models.Task{}
+	if err := c.Bind(&task); err != nil {
+		return err
+	}
+
+	q := tx.Where("finished_at is null")
+	count, err := q.Count(task)
+	if err != nil {
+		return err
+
+	}
+
+	countInfoIncomplete := fmt.Sprintf("%v Incomplete Tasks", count)
+
+	c.Set("count", countInfoIncomplete)
+
+	verrs, err := tx.ValidateAndCreate(&task)
+	if err != nil {
+		return err
+	}
+	if verrs.HasAny() {
+		c.Set("error", verrs.Get("name"))
+		return c.Render(http.StatusUnprocessableEntity, r.HTML("new_task.plush.html"))
+	}
+
+	return c.Redirect(http.StatusSeeOther, "/table-incomplete")
+}
+
+func Delete(c buffalo.Context) error {
+
+	tasks := &models.Task{}
+	tx, ok := c.Value("tx").(*pop.Connection)
+	if !ok {
+		return errors.New("no transaction found")
+	}
+
+	if err := tx.Find(tasks, c.Param("task_id")); err != nil {
+		return c.Error(http.StatusNotFound, err)
+	}
+
+	if err := tx.Destroy(tasks); err != nil {
+		return err
+	}
+
+	return c.Redirect(302, "/table-incomplete")
+}
+func ShowEdit(c buffalo.Context) error {
+
+	tasks := &models.Task{}
+
+	tx, ok := c.Value("tx").(*pop.Connection)
+	if !ok {
+		return errors.New("no transaction found")
+	}
+
+	q := tx.Where("finished_at is null")
+	count, err := q.Count(tasks)
+	if err != nil {
+		return err
+	}
+
+	countInfoIncomplete := fmt.Sprintf("%v Incomplete Tasks", count)
+
+	c.Set("count", countInfoIncomplete)
+
+	if err := tx.Find(tasks, c.Param("task_id")); err != nil {
+		return c.Error(http.StatusNotFound, err)
+	}
+
+	c.Set("task", tasks)
+
+	return c.Render(http.StatusOK, r.HTML("edit_task.plush.html"))
+}
+
+func Update(c buffalo.Context) error {
+
+	tasks := models.Task{}
+
+	tx, ok := c.Value("tx").(*pop.Connection)
+	if !ok {
+		return errors.New("no transaction found")
+	}
+
+	if err := tx.Find(&tasks, c.Param("task_id")); err != nil {
+		return c.Error(http.StatusNotFound, err)
+	}
+
+	if err := c.Bind(&tasks); err != nil {
+		return err
+	}
+	q := tx.Where("finished_at is null")
+	count, err := q.Count(tasks)
+	if err != nil {
+		return err
+
+	}
```

Comment:
Can you move this into a method to reuse it?

Reply:
Okey 👍🏻